android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/WorkerBinder.kt

```diff
@@ -39,9 +40,22 @@ object WorkerBinder {
    * @param worker The class that wants to be informed when to start and stop doing work.
    * @return [WorkerUnbinder] to unbind [Worker&#39;s][Worker] lifecycle.
    */
+  @OptIn(DelicateCoroutinesApi::class)
   @JvmStatic
-  open fun bind(interactor: Interactor<*, *>, worker: Worker): WorkerUnbinder {
-    return bind(mapInteractorLifecycleToWorker(interactor.lifecycle()), worker)
+  public fun bind(interactor: Interactor<*, *>, worker: Worker): WorkerUnbinder {
+    val workerEvents = interactor.lifecycleFlow.map { it.asWorkerEvent() }
+    /*
+     * We need `Dispatchers.Unconfined` to react immediately to lifecycle flow emissions, and we need
+     * `CoroutineStart.Undispatched` to prevent coroutines launched in `onStart` with `Dispatchers.Unconfined`
+     * from forming an event loop instead of starting eagerly.
+     */
+    val job = GlobalScope.launch(RibDispatchers.Unconfined, start = CoroutineStart.UNDISPATCHED) {
```

Comment:
why we need GlobalScope? wouldn't this introduce a risk to potential leaks?

Reply:
In theory it would, but we are using "takeWhile" on flow that makes it terminate once the interactor lifecycle terminates (so the coroutine also terminates)

The reason for not using `interactor.coroutineScope` is to keep retro-compatibily with calling bind before `didBecomeActive` which should be rare, but was supported by previous implementation nevertheless

Reply:
I see, thanks for clarifying